plugins/org.yakindu.sct.doc.user/src/user-guide/embedded_systems_integration_guide.textile

```diff
@@ -0,0 +1,281 @@
+
+bc.. 
+int main() {
+	while(true) {
+		if(bool_flag1 == true) {
+			doStuff();
+			bool_flag1 = false;
+		}
+		if(bool_flag2 == true) {
+			doDiffferentStuff();
+			bool_flag2 = false;
+		}
+		// sleep_mode();
+	}
+}
+
+ISR_1 {
+	bool_flag1 = true;
+}
+
+ISR_2 {
+	bool_flag2 = true;
+}
+
+p. 
+Each ISR uses one specific boolean flag, which stores wherever an interrupt has occurred. This pattern is often combined with a sleep mode, so that the microcontroller's CPU is only active after an interrupt. This pattern is often used for low power applications, which are powered by batteries.
+
+h2(#Example-State-Machine-for-Integration). Example State Machine for Integration
+
+The complexity of the state machine can raise in a development process. Thus, it's recommended to specify of how the API (Applicable Programming Interface) should be accessed. While the API depends on the execution and used event types, this guide tries to break down the main elements and define a expandable design pattern. Therefore, the following example will be used for all integration guides.
+!(large)images/images_embedded_systems_integration_guide/platform_integration_example.jpeg(Example State Machine)!
+
+p=. Example State Machine
+
+The state machine contains three types of events: two __in events__ (inEvent1, inEvent2), two __out events__ (outEvent1, outEvent2) and one __timed event__ (after 2 s). Two states are connected with the inEvent1 and the after 2 s event. Each state raises one of __out event__. Additionally, a final state has been added, which also is part of the API. It only can be reached by raising the __in event__ inEvent2, if the state machine's __state__ StateA is active.
+
+These are the main parts of the state machine, which can be used and must be handled:
+* raise in events
+* raise timed events
+* handle out events
+* check if the state machine is final
+
+h2(#Bare-metal-cycle-based-integration). Bare-Metal Integration
+
+This section describes how to integrate a state machine with a cycle based or event driven execution semantic on any microcontroller, so on any embedded device. The integration can be structured in three parts, which results the main part of your program.
+* Initialization - called once
+* While loop - calling the state machine until it's final
+* Final - called if the state machine is final
+The execution of the program can be visualized as a state machine:
+
+!(large)images/images_embedded_systems_integration_guide/platform_integration_simple.png(High level description of bare metal cycle based state machine integration)!
+
```

Comment:
I won't use a meta state machine to describe the code structure here. It could be confusing to introduce two state machines on two different meta levels. The overall structure already became clear from the coding pattern described above.

Reply:
:+1: